npc/re/instances/FridayDungeon.txt

```diff
@@ -0,0 +1,782 @@
+							if(isbegin_quest(12379)<1) setquest 12379;
+					}
+					end;
+				case 3:
+					mes "[Marry Jay]";
+					mes "You have not prepared?";
+					mes "Can't really help it I guess";
+					close;
+			}
+		} else {
+			switch(select("Enter Friday Memorial:Cancel")) {
+				case 1:
+					switch( instance_enter(.@md_name$) ) {
+						case IE_NOMEMBER:
+							mes "Only party members can enter this Memorial dungeon.";
+							close;
+						case IE_NOINSTANCE:
+							mes "The Friday Memorial Dungeon does not exist.";
+							mes "Your party leader has not yet created the Memorial dungeon.";
+							close;
+						case IE_OTHER:
+							mes "An unknown error occurred.";
+							close;
+						case IE_OK:
+							mes "[Marry Jay]";
+							mes "Off you go!";
+							mapannounce "gef_tower", "" + strcharinfo(0) + " of the party, " + getpartyname( getcharid(1) ) + ", is entering the " + .@md_name$ + ".", bc_map, 0xFF99;
+							if(isbegin_quest(12379)<1) setquest 12379;
+							end;
+					}
+					break;
+				case 2:
+					mes "[Marry Jay]";
+					mes "You have not prepared?";
+					mes "Can't really help it I guess";
+					close;
+			}
+		}
+	end;
+}
+
+//Instance
+1@md_gef,185,81	monster	Shining Plant	1083,1,1800000,900000
+
+1@md_gef,110,129,6	script	Marry Jay#0_1	4_F_JOB_BLACKSMITH,{
+
+	if (is_party_leader() == true) {
+		mes "[Marry Jay]";
+		mes "Before we start setting up the environment here, we'll give you the chance to choose an appropriate difficulty level.";
+		mes "I could make my own choice, but each person handles it differently based on their abilities.";
+		next;
+		switch(select("Cancel:LV130 ~ 199:LV200+ ")) {
+			case 1:
+				close;
+			//MinLevel, Lich Lord, Nightmare, Jakk, Ghoul, Drainliar
+			case 2:
+				setarray 'mob,130,3658,3660,3662,3664,3666; break;
+			case 3:
+				if(BaseLevel < 200){
+				mes "[Marry Jay]";
+					mes "You can't choose this difficulty, because your base level is below 200.";
+					close;
+				}
+				'hard = 1;
+				setarray 'mob,200,3659,3661,3663,3665,3667; break;
+		}
+		if(BaseLevel < 'mob[0]){
+			mes "[Marry Jay]";
+			mes "Your base level is too low to select this, you need to be level "+'mob[0]+" or above";
+			close;
+		}
+		close2;
+		npctalk "Now, this place is about to become a hunting ground. I'll get out and close the front door.";
+		sleep 4000;
+		npctalk "Good Luck~";
+		sleep 3000;
+		monster 'map$,0,0,"Nightmare",'mob[2],20,instance_npcname(strnpcinfo(0))+"::OnNightmareDead";
+		monster 'map$,0,0,"Jakk",'mob[3],30,instance_npcname(strnpcinfo(0))+"::OnJakkDead";
+		monster 'map$,0,0,"Ghoul",'mob[4],30,instance_npcname(strnpcinfo(0))+"::OnGhoulDead";
+		monster 'map$,0,0,"Drainliar",'mob[5],20,instance_npcname(strnpcinfo(0))+"::OnDrainliarDead";
+		hideonnpc instance_npcname(strnpcinfo(0));
+		enablenpc instance_npcname("The Stranger#cadaver");
+		enablenpc instance_npcname("Bizzare Sculpture#boss");
```

Comment:
to prevent multiple spawning if the leader retalk within the time limit
```suggestion
		if ('step == 0) {
			'step = 1;
			monster 'map$,0,0,"Nightmare",'mob[2],20,instance_npcname(strnpcinfo(0))+"::OnNightmareDead";
			monster 'map$,0,0,"Jakk",'mob[3],30,instance_npcname(strnpcinfo(0))+"::OnJakkDead";
			monster 'map$,0,0,"Ghoul",'mob[4],30,instance_npcname(strnpcinfo(0))+"::OnGhoulDead";
			monster 'map$,0,0,"Drainliar",'mob[5],20,instance_npcname(strnpcinfo(0))+"::OnDrainliarDead";
			hideonnpc instance_npcname(strnpcinfo(0));
			enablenpc instance_npcname("The Stranger#cadaver");
			enablenpc instance_npcname("Bizzare Sculpture#boss");
		}
```